tools/clang/unittests/HLSLExec/ExecutionTest.cpp

```diff
@@ -11096,6 +11228,609 @@ TEST_F(ExecutionTest, PackUnpackTest) {
+typename = std::enable_if_t<std::is_arithmetic_v<T>>>
+struct SLongVectorBinaryOp {
+  T scalarInput;
+  std::array<T, N> vecInput1;
+  std::array<T, N> vecInput2;
+  std::array<T, N> vecOutput;
+};
+
+// vec1 == Expected vector
+// vec2 == Actual vector
+template <typename T, typename = std::enable_if_t<std::is_arithmetic_v<T>>>
+bool DoVectorsMatch(const std::vector<T>& vec1, const std::vector<T>& vec2, double tolerance) {
+    // Ensure both vectors have the same size
+    if (vec1.size() != vec2.size()) {
+      VERIFY_IS_TRUE(false, L"Vectors are of different sizes!");
+      return false;
+    }
+    
+    bool vectorsMatch = true;
+
+    // Stash mismatched indexes for easy failure logging later
+    std::vector<size_t> mismatchedIndexes;
+    for (size_t i = 0; i < vec1.size(); ++i) {
+      if (tolerance == 0 && vec1[i] != vec2[i]) {
+          vectorsMatch = false;
+          mismatchedIndexes.push_back(i); 
+      } else if constexpr (std::is_same_v<T, bool>) {
+        // Compiler was very picky and wanted an explicit case for bools to
+        // avoid a warning about comparing bools with >.
+        if (vec1[i] != vec2[i]) {
+            vectorsMatch = false;
+            mismatchedIndexes.push_back(i); 
+        }
+      } else {
+        T diff = vec1[i] > vec2[i] ? vec1[i] - vec2[i] : vec2[i] - vec1[i];
+        if (diff > tolerance) {
+          vectorsMatch = false;
+          mismatchedIndexes.push_back(i);
+        }
+      }
+    }
+
+    // Print the mismatched indexes and their corresponding elements
+    if (!mismatchedIndexes.empty()) {
+        for (size_t index : mismatchedIndexes) {
+          std::wstringstream wss(L"");
+          wss << L"Mismatch at Index: " << index;
+          wss << L" Expected Value:" << vec1[index];
+          wss << L" Actual Value:" << vec2[index];
+          WEX::Logging::Log::Error(wss.str().c_str());
+        }
+    }
+
+    return vectorsMatch;
+}
+
+template <typename T, typename = std::enable_if_t<std::is_arithmetic_v<T>>>
+T parseStringToNumber(std::wstring& str) {
+  if constexpr (std::is_same_v<T, uint16_t>) {
+    return static_cast<uint16_t>(std::stoul(str));
+  } else if constexpr (std::is_same_v<T, uint32_t>) {
+    return std::stoul(str);
+  } else if constexpr (std::is_same_v<T, uint64_t>) {
+    return std::stoull(str);
+  } else if constexpr (std::is_same_v<T, int16_t>) {
+    return static_cast<int16_t>(std::stol(str));
+  } else if constexpr (std::is_same_v<T, int32_t>) {
+    return std::stol(str);
+  } else if constexpr (std::is_same_v<T, int64_t>) {
+    return std::stoll(str);
+  } else if constexpr (std::is_same_v<T, float>) {
+    return std::stof(str);
+  } else if constexpr (std::is_same_v<T, double>) {
+    return std::stod(str);
+  } else if constexpr (std::is_same_v<T, bool>) {
+    bool boolVal = std::stod(str) > 0.0 ? true : false;
+    return boolVal ;
+  } else {
+    VERIFY_IS_TRUE(false, L"Unsupported type for parsing string to number.");
+  }
```

Comment:
nit: applying two LLVM coding conventions: 
* [don't use `else` after a `return`](https://llvm.org/docs/CodingStandards.html#don-t-use-else-after-a-return)
* [don't use braces for single statement bodies](https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements)
```suggestion
  if constexpr (std::is_same_v<T, uint16_t>)
    return static_cast<uint16_t>(std::stoul(str));
   if constexpr (std::is_same_v<T, uint32_t>)
    return std::stoul(str);
  if constexpr (std::is_same_v<T, uint64_t>)
    return std::stoull(str);
  if constexpr (std::is_same_v<T, int16_t>)
    return static_cast<int16_t>(std::stol(str));
  if constexpr (std::is_same_v<T, int32_t>)
    return std::stol(str);
  if constexpr (std::is_same_v<T, int64_t>)
    return std::stoll(str);
  if constexpr (std::is_same_v<T, float>)
    return std::stof(str);
  if constexpr (std::is_same_v<T, double>)
    return std::stod(str);
  if constexpr (std::is_same_v<T, bool>) {
    bool boolVal = std::stod(str) > 0.0 ? true : false;
    return boolVal ;
  }
  VERIFY_IS_TRUE(false, L"Unsupported type for parsing string to number.");
```